runtime/ts/isolate/sanitize.ts

```diff
@@ -0,0 +1,128 @@
+import * as dompurify from 'dompurify';
+
+/*
+ * Ugly workaround. Internal DOM operations within DOMPurify fail because style% is an illegal attribute name.
+ * Browsers allow it when using innerHTML to construct the DOM from a string, but disallow '%' in attribute names when
+ * calling element.setAttribute().
+ */
+const ESCAPED_STYLE_PCT = "__style__pct__";
+
+/*
+ * DOMPurify must be told about non-standard HTML elements (e.g. web components). For now we just whitelist them here,
+ * but we should consider having some mechanism in the runtime to auto-discover these. Note, if you add a new
+ * Web Component to the shell, it must be registered here.
+ */
+const CUSTOM_ELEMENTS: string[] = [
+    'app-shell',
```

Comment:
None of the Shell elements should be allowed by Particles. There is a much shorter list of elements specifically whitelisted for use in Particle DOM. Currently, these are collated for import [here](https://github.com/PolymerLabs/arcs/blob/master/shell/apps/common/whitelisted.js). Probably we should have this import supply only the paths and make it the source of truth for the sanitizer and a Shell loader.